website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+    credentials on disk and use those to authenticate to Vault, that would not
+    be recommended.)
+  * If you have non-EC2 instance entities, such as IAM users, Lambdas in IAM
+    roles, or developer laptops using [AdRoll's Hologram](https://github.com/AdRoll/hologram)
+    then you would need to use the AWS-IAM backend.
+  * If you have EC2 instances which are already in an IAM instance profile, then
+    you could use either backend.
+
+## Authentication Workflow
+
+AWS has introduced the [`sts:GetCallerIdentity`](http://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html)
+API method to allow you to validate the identity of a caller. The client signs
+a `GetCallerIdentity` query using the [AWS Signature v4
+algorithm](http://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html) and
+submits that signed query to the Vault server. The Vault server then forwards
+it on to the AWS STS service and validates the result back. Clients don't even
+need network-level access to talk to the AWS STS API endpoint; they merely need
+to sign the credentials. However, it means that the Vault server DOES need
+network-level access to send requests to the STS endpoint.
+
+Each signed AWS request includes the current timestamp to mitigate the risk of
+replay attacks. In addition, Vault allows you to require an additional header,
+`X-Vault-Server`, to be present to mitigate against different types of replay
+attacks (such as a signed `GetCallerIdentity` request stolen from a dev Vault
+instance and used to authenticate to a prod Vault instance). Vault further
+requires that this header be one of the headers included in the AWS signature
+and relies upon AWS to authenticate that signature.
+
+While the AWS API endpoints support both signed GET and POST requests, for
+simplicity, the aws-iam backend supports only POST requests.
+
+## Authorization Workflow
+
+Roles are mapped from AWS IAM principals back to Vault roles. At present, two
+types of principals are supported: AWS IAM Users and AWS IAM roles. For users,
+you simply bind the full ARN of the user, e.g., `arn:aws:iam::123456789012:user/MyUserName`
+(where `123456789012` is your AWS account number nad `MyUserName` is your IAM
+username). Assumed roles are a little tricky: when authenticating with IAM roles,
+you are simultaneously _two_ different principals: `arn:aws:iam::123456789012:role/MyRoleName`
+*and* `arn:aws:sts::123456789012:assumed-role/MyRoleName/RoleSessionName`. In
+this case, you should bind the former ARN to your Vault roles.
+
+## Authentication
+
+### Via the CLI
+
+#### Enable AWS IAM authentication in Vault.
+
+```
+$ vault auth-enable aws-iam
+```
+
+#### Configure the policies on the role.
+
+```
+$ vault write auth/aws-iam/role/dev-role bound_iam_principal=arn:aws:iam::123456789012:role/my_role policies=prod,dev max_ttl=500h
+```
+
+#### Configure a required X-Vault-Server Header (recommended)
```

Comment:
I'd love some feedback. It would be great to require this header be set, but I'm not sure what the right default is. redirect_addr seems like a promising start, but I'm not sure if there's a good way to access that configuration variable from within an authentication backend. Any thoughts?